resources/sagemaker-notebookinstancelifecycleconfigs.go

```diff
@@ -0,0 +1,68 @@
+package resources
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/sagemaker"
+	"github.com/rebuy-de/aws-nuke/pkg/types"
+)
+
+type SageMakerNotebookInstanceLifecycleConfig struct {
+	svc  *sagemaker.SageMaker
+	Name *string
+}
+
+func init() {
+	register("SageMakerNotebookInstanceLifecycleConfig", ListSageMakerNotebookInstanceLifecycleConfigs)
+}
+
+func ListSageMakerNotebookInstanceLifecycleConfigs(sess *session.Session) ([]Resource, error) {
+	svc := sagemaker.New(sess)
+	resources := []Resource{}
+
+	params := &sagemaker.ListNotebookInstanceLifecycleConfigsInput{
+		MaxResults: aws.Int64(30),
+	}
+
+	for {
+		resp, err := svc.ListNotebookInstanceLifecycleConfigs(params)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, lifecycleConfig := range resp.NotebookInstanceLifecycleConfigs {
+			resources = append(resources, &SageMakerNotebookInstanceLifecycleConfig{
+				svc:  svc,
+				Name: lifecycleConfig.NotebookInstanceLifecycleConfigName,
+			})
+		}
+
+		if resp.NextToken == nil {
+			break
+		}
+
+		params.NextToken = resp.NextToken
+	}
+
+	return resources, nil
+}
+
+func (f *SageMakerNotebookInstanceLifecycleConfig) Remove() error {
+
+	_, err := f.svc.DeleteNotebookInstanceLifecycleConfig(&sagemaker.DeleteNotebookInstanceLifecycleConfigInput{
+		NotebookInstanceLifecycleConfigName: f.Name,
+	})
+
+	return err
+}
+
+func (f *SageMakerNotebookInstanceLifecycleConfig) String() string {
+	return *f.Name
+}
+
+func (f *SageMakerNotebookInstanceLifecycleConfig) Properties() types.Properties {
+	properties := types.NewProperties()
+	properties.
+		Set("NotebookInstanceLifecycleConfigName", f.Name)
```

Comment:
```suggestion
		Set("Name", f.Name)
```

Hello. Sorry for not being clear about this, but we were talking about both names. Since the property is only available for the `SageMakerNotebookInstanceLifecycleConfig` resource, there is no need to add the whole prefix to the property.